src/lib/GridController.ts

```diff
@@ -0,0 +1,27 @@
+import { getAvailablePosition } from './utils/grid';
+import type { GridParams } from './types';
+
+export class GridController {
+	static gridParams: GridParams | undefined;
+	static getFirstAvailablePosition(w: number, h: number) {
```

Comment:
Don't make these methods static. 
This makes it impossible to use two or more grids on same page, like:
```svelte
<script>
	let galleryController;

	let secondController;

	// now galleryController uses same params as secondController
</script>

<Grid bind:controller={galleryController} />

<Grid bind:controller={secondController} />
```

Just use constructor:
```ts
// GridController.ts

export class GridController {
	gridParams: GridParams;
	
	constructor(gridParams: GridParams) {
		this.gridParams = gridParams;
	}

	getFirstAvailablePosition(w: number, h: number) {
		/** */
	}
```

```svelte
// Grid.svelte
<script>
	// ...
	let gridController = GridController($gridSettings);

	$: gridController.gridParams = $gridSettings;
	// ...
</script>


<!-- -->
	<slot {gridController} />
<!-- -->
```